FirebaseAppCheck/Tests/Unit/AppAttestProvider/FIRAppAttestKeyIDStorageTests.m

```diff
@@ -54,38 +55,88 @@ - (void)testSetAndGetAppAttestKeyID {
+      [[FIRAppAttestKeyIDStorage alloc] initWithAppName:@"FIRAppAttestKeyIDStorageTestsApp2"
+                                                  appID:self.appID];
 
+  // 2. Store an app attest key ID in an app's app attest key ID storage.
+  NSString *appAttestKeyID = @"app_attest_key_ID";
   FBLPromise *setPromise = [self.storage setAppAttestKeyID:appAttestKeyID];
+  XCTAssert(FBLWaitForPromisesWithTimeout(0.5));
   XCTAssertEqualObjects(setPromise.value, appAttestKeyID);
   XCTAssertNil(setPromise.error);
 
-  // 2. Try to read the app attest key ID in another app's app attest key ID storage.
-  FIRAppAttestKeyIDStorage *storage2 =
-      [[FIRAppAttestKeyIDStorage alloc] initWithAppName:self.appName appID:@"app_id_2"];
-  __auto_type getPromise = [storage2 getAppAttestKeyID];
-  XCTAssertNotNil(getPromise.error);
-  XCTAssertEqualObjects(getPromise.error, [FIRAppCheckErrorUtil appAttestKeyIDNotFound]);
+  // 3. Try to read the app attest key ID from the other app storages.
+  __auto_type getPromise2 = [storage2 getAppAttestKeyID];
+  XCTAssert(FBLWaitForPromisesWithTimeout(0.5));
+  XCTAssertNotNil(getPromise2.error);
+  XCTAssertEqualObjects(getPromise2.error, [FIRAppCheckErrorUtil appAttestKeyIDNotFound]);
+
+  __auto_type getPromise3 = [storage3 getAppAttestKeyID];
+  XCTAssert(FBLWaitForPromisesWithTimeout(0.5));
+  XCTAssertNotNil(getPromise3.error);
+  XCTAssertEqualObjects(getPromise3.error, [FIRAppCheckErrorUtil appAttestKeyIDNotFound]);
+
+  // 4. Assert that storages can be updated independently.
+  NSString *appAttestKeyID_2 = @"app_attest_key_ID_2";
+  __auto_type setPromise2 = [storage2 setAppAttestKeyID:appAttestKeyID_2];
+  XCTAssert(FBLWaitForPromisesWithTimeout(0.5));
+  XCTAssertEqualObjects(setPromise2.value, appAttestKeyID_2);
+  XCTAssertNil(setPromise2.error);
+
+  NSString *appAttestKeyID_3 = @"app_attest_key_ID_3";
+  __auto_type setPromise3 = [storage3 setAppAttestKeyID:appAttestKeyID_3];
+  XCTAssert(FBLWaitForPromisesWithTimeout(0.5));
+  XCTAssertEqualObjects(setPromise3.value, appAttestKeyID_3);
+  XCTAssertNil(setPromise3.error);
+
+  __auto_type getPromiseForAppAttestKeyID = [self.storage getAppAttestKeyID];
+  XCTAssert(FBLWaitForPromisesWithTimeout(0.5));
+  XCTAssertNil(getPromiseForAppAttestKeyID.error);
+
+  __auto_type getPromiseForAppAttestKeyID_2 = [storage2 getAppAttestKeyID];
+  XCTAssert(FBLWaitForPromisesWithTimeout(0.5));
+  XCTAssertNil(getPromiseForAppAttestKeyID_2.error);
+
+  __auto_type getPromiseForAppAttestKeyID_3 = [storage3 getAppAttestKeyID];
+  XCTAssert(FBLWaitForPromisesWithTimeout(0.5));
+  XCTAssertNil(getPromiseForAppAttestKeyID_3.error);
+
+  XCTAssertNotEqualObjects(getPromiseForAppAttestKeyID.value, getPromiseForAppAttestKeyID_2.value);
+  XCTAssertNotEqualObjects(getPromiseForAppAttestKeyID_2.value,
+                           getPromiseForAppAttestKeyID_3.value);
+
+  // 5. Cleanup other storages.
+  [storage2 setAppAttestKeyID:nil];
+  [storage3 setAppAttestKeyID:nil];
```

Comment:
optional: now it looks a bit harder to follow the test. We probably may slightly improve readability here, e.g.:
- create a helper method like `- assertIndependentSetGetForStoragesWithAppName1:appID1:appName2:appID2:` where we assert independent reads and writes
- call this method with all combinations we are interested in

WDYT?

Reply:
Thanks for taking a look. Yeah, I agree. I was thinking about how to make it easier to understand. I like this approach ^ though. 👍